purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -1557,7 +1564,10 @@ class Purchases internal constructor(
                 offeringIdentifier,
                 listener
             )
-        } ?: listener.dispatch(PurchasesError(PurchasesErrorCode.OperationAlreadyInProgressError).also { errorLog(it) })
+        } ?: run {
+            listener.dispatch(PurchasesError(PurchasesErrorCode.OperationAlreadyInProgressError).also { errorLog(it) })
+            getAndClearProductChangeCallback()
```

Comment:
I think it would make sense to call this before listener.dispatch? Just in case the listener blocks the thread or something and another purchase comes through

Reply:
I guess dispatch changes the thread anyways, but do you think it might be better to call this before the dispatch?

Reply:
yeah, good point!